library/list.lisp

```diff
@@ -234,6 +235,16 @@
     "Like INDEX, but errors if the index is not found."
     (from-some "There is no NTH" (index n l)))
 
+  (declare nth-cdr (Ufix -> List :a -> (Optional (List :a))))
```

Comment:
To be consistent with CDR this ought to just do CDR at most N times, shorting out when it reaches the empty list. 

Alternatively, to be consistent with NTH, we could error on (nth-cdr n Nil) for n>0. I don't recommend it though.